osu.Game/Screens/Edit/Compose/Components/DistanceSnapGrid.cs

```diff
@@ -146,5 +146,20 @@ protected ColourInfo GetColourForIndexFromPlacement(int placementIndex)
             int repeatIndex = placementIndex / beatDivisor.Value;
             return ColourInfo.SingleColour(colour).MultiplyAlpha(0.5f / (repeatIndex + 1));
         }
+
+        public float GetBeatSnapDistanceAt(HitObject referenceObject) => (float)(snapProvider.GetBeatSnapDistanceAt(referenceObject) * distanceSpacingMultiplier.Value);
```

Comment:
While the implementation looks correct, I believe this is potentially implemented in the wrong place.

Right now, the DI'd `snapProvider` here is `HitObjectComposer` which directly owns the `DistanceSnapGrid`. In other implementations (like https://github.com/ppy/osu/blob/bb086800b1e528205cfcd635c093798bc87e2a0c/osu.Game.Rulesets.Osu/Edit/OsuHitObjectComposer.cs#L142-L143) the application of the grid snap is being done in `HitObjectComposer` itself, and I feel like maybe this should be moved there as well?

The rationale is that other components may call `GetBeatSnapDistanceAt` on their resolved snap provider, which will be the `HitObjectComposer`. I think even in those cases the distance spacing multiplier needs to be accounted for.

Not 100% sure and maybe you'll find a reason it can't be done there, but worth a try (mainly to reduce complexity if it does so).

Reply:
To confirm, you're saying that "the only time we will want to apply distance snapping is when interacting with the distance spacing grid"?

Reply:
When the distance snap/spacing grid is enabled, correct.